modules/structs/hook.go

```diff
@@ -24,6 +24,7 @@ type Hook struct {
 	Events              []string          `json:"events"`
 	AuthorizationHeader string            `json:"authorization_header"`
 	Active              bool              `json:"active"`
+	IsSystemWebhook     bool              `json:"is_system_webhook"`
```

Comment:
Although the database stored as the name, but maybe we can have a better name like `IsSystemOrDefault`? Or even `ScopeType`?

Reply:
I've gotten around to this again finally.

I decided to move `IsSystemOrDefault`/`IsSystemWebhook` into a path parameter:

https://github.com/go-gitea/gitea/blob/5384c1d877a99e8fb3d5be148643490ad2afd68d/routers/api/v1/api.go#L1384-L1390

https://github.com/go-gitea/gitea/blob/5384c1d877a99e8fb3d5be148643490ad2afd68d/routers/api/v1/admin/hooks.go#L24

https://github.com/go-gitea/gitea/blob/5384c1d877a99e8fb3d5be148643490ad2afd68d/routers/api/v1/admin/hooks.go#L62

https://github.com/go-gitea/gitea/blob/5384c1d877a99e8fb3d5be148643490ad2afd68d/routers/api/v1/admin/hooks.go#L107

https://github.com/go-gitea/gitea/blob/5384c1d877a99e8fb3d5be148643490ad2afd68d/routers/api/v1/admin/hooks.go#L139

https://github.com/go-gitea/gitea/blob/5384c1d877a99e8fb3d5be148643490ad2afd68d/routers/api/v1/admin/hooks.go#L178


Doing this ballooned the PR, but it makes the API more friendly:

![Screenshot 2023-07-01 at 23-41-57 Gitea API](https://github.com/go-gitea/gitea/assets/987487/ff30e3e2-49c7-4bcc-bab9-1b18779eac55)

and it also matches how the admin web UI works, where the two links on `/admin/system-hooks/`

![Screenshot 2023-07-01 at 23-43-17 Gitea Git with a cup of tea](https://github.com/go-gitea/gitea/assets/987487/40c8f311-9a58-4a04-bec9-345dd148ab65)

go to `/admin/system-hooks/{kind}/new` and `http://localhost:3000/admin/default-hooks/{kind}/new` respectively.

What do you think?